sqlg-core/src/main/java/org/umlg/sqlg/structure/SqlgDataSourceFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.umlg.sqlg.structure;
+
+import org.apache.commons.configuration.Configuration;
+import javax.sql.DataSource;
+
+/**
+ * Created by petercipov on 27/02/2017.
+ */
+public interface SqlgDataSourceFactory {
+
+    SqlgDataSource setup(String driver, final Configuration configuration) throws Exception;
+    SqlgDataSource setupFromJndi(String jndiName, Configuration configuration) throws Exception;
```

Comment:
The way the original code "handled" it is by having a single "if" in the SqlgGraph instatiation where it checked whether the `jdbc.url` configuration property started with `jndi:`. If so, it used "special handling" of that URL such that it read the datasource from the JNDI tree and placed the obtained datasource in the datasource cache under the value of `jdbc.url`. I.e. the cache would contain a key  "jndi:...." with the datasource obtained from JNDI.

What you have now is the ability to obtain an SqlgDatasource by a JNDI name and then on that you are able to call `get("some_jdbc_url")`. That is confusing, because the API suggests that you should be able to get different datasources for different supplied urls from the SqlgDatasource instance. That is not true for the JNDI-based datasources.